docs/source/deployment/kubernetes/architecture-overview.rst

```diff
@@ -0,0 +1,144 @@
+
+Connection sequence
+~~~~~~~~~~~~~~~~~~~
+
+In order to fully understand how this architecture is used, it is time to analyse how the different components communicate.
+Let us start off with the Float component, one of the two modes of the Corda Firewall.
+The Float component starts off in a mode where it is just waiting for a Bridge component to take control of it.
+In this mode, it is only listening on one port, the port where we expect the Bridge to connect to.
+The connection between the Bridge and the Float is mutually authenticated TLS using certificates that originate from a shared trust root.
+As the Bridge connects to the the Float component on its listening port, the TLS handshake occurrs and verifies that both the identities of the two components are as expected, but also that the trust root they both use is the same and is valid.
+Once the Bridge has taken control of the Float, the Float will enter into the listening mode, where it will open up a port for listening for external AMQP over TLS connections (the peer to peer (p2p) traffic).
+At this point the Corda Firewall component is fully started and established, ready to communicate with other nodes on the network.
+
+Next we will look at the Bridge component.
+Before the Bridge connects to the Float as mentioned above, the Bridge will connect to the Artemis message queue, whether the Artemis message queue is in-process in the Corda Node or running out-of-process as a stand-alone Artemis MQ does not matter, the Bridge initiates the connection.
+This connection is using mutually authenticated TLS as was the connection between the Bridge and the Float.
+
+If we are having Artemis running out-of-process we have one additional step still, where the actual Corda Node needs to connect to the Artemis service as well. This connection works just like the Bridge, using mutually authenticated TLS connection originating from the Corda Node to the Artemis messaging server (messaging queue, MQ). This should be set up using a shared trust root for the TLS certificates of the components.
+
+At this point we have established the full connection sequence. We have:
+
+* The Corda Node connected to the Artemis MQ
+* Artemis MQ connected to the Bridge
+* Bridge connected to the Float
+* Optionally Bridge connected to SOCKS and HTTP proxy
+
+Now we can send and receive messages to/from other nodes on the network.
+
+Internal Kubernetes cluster details
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Let's start by diving into the details of the internal zones Kubernetes cluster.
+The main components, the Bridge, Artemis and the Node will all have Kubernetes services to handle port mapping and then Kubernetes deployments which set how many replicas (pods) should be deployed.
```

Comment:
Suggestion: this feels like a long paragraph. Break it up a little for the reader. 

Let’s start by diving into the details of the internal zones Kubernetes cluster. 

The main components, the Bridge, Artemis and the Node will all have Kubernetes services to handle port mapping and then Kubernetes deployments which set how many replicas (pods) should be deployed. 

Bridge is currently using one replica running at any given time, if it fails it is rescheduled and reconnects to the Float on startup. The reason why we only use one at this point in time, is because there is currently only one Bridge per Float. 

Artemis is currently not implemented in the related deployment scripts, but is envisioned to use two replicas where one is master and the other one will be the failover one, should master go down for any reason. 

The Corda Node is using one replica always, because currently Corda Node only supports Hot-Cold HA, which means you cannot have two versions running at the same time, connected to the same database. This may change in future versions of Corda and in that case this decision may be revisited. In the meantime however, we can rely on Kubernetes to automatically restart a failed pod and self-heal.